Dockerfile

```diff
@@ -1,74 +1,117 @@
-# for now.
-EXPOSE 3000
+RUN useradd --system --create-home --shell /usr/sbin/nologin stf
+
+FROM nodebase as with_packages
 
-# Install app requirements. Trying to optimize push speed for dependant apps
-# by reducing layers as much as possible. Note that one of the final steps
-# installs development files for node-gyp so that npm install won't have to
-# wait for them on the first native module installation.
+# Install additional packages for building things
 RUN export DEBIAN_FRONTEND=noninteractive && \
-    useradd --system \
-      --create-home \
-      --shell /usr/sbin/nologin \
-      stf-build && \
-    useradd --system \
-      --create-home \
-      --shell /usr/sbin/nologin \
-      stf && \
-    sed -i'' 's@http://archive.ubuntu.com/ubuntu/@mirror://mirrors.ubuntu.com/mirrors.txt@' /etc/apt/sources.list && \
-    apt-get update && \
-    apt-get -y install wget python build-essential && \
-    cd /tmp && \
-    wget --progress=dot:mega \
-      https://nodejs.org/dist/v8.9.3/node-v8.9.3-linux-x64.tar.xz && \
-    tar -xJf node-v*.tar.xz --strip-components 1 -C /usr/local && \
-    rm node-v*.tar.xz && \
-    su stf-build -s /bin/bash -c '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js install' && \
-    apt-get -y install libzmq3-dev libprotobuf-dev git graphicsmagick openjdk-8-jdk yasm && \
-    apt-get clean && \
-    rm -rf /var/cache/apt/* /var/lib/apt/lists/* && \
-    mkdir /tmp/bundletool && \
-    cd /tmp/bundletool && \
-    wget --progress=dot:mega \
-      https://github.com/google/bundletool/releases/download/1.2.0/bundletool-all-1.2.0.jar && \
-    mv bundletool-all-1.2.0.jar bundletool.jar
-
-# Copy app source.
-COPY . /tmp/build/
-
-# Give permissions to our build user.
-RUN mkdir -p /app && \
-    chown -R stf-build:stf-build /tmp/build /tmp/bundletool /app
-
-# Switch over to the build user.
-USER stf-build
-
-# Run the build.
+    apt-get -y --no-install-recommends install build-essential git yasm jq python vim
+
+# Install node-gyp ahead of time to avoid installation on native module install
+# RUN /bin/bash -c '/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js install'
+
+# Install just the package dependencies before copying in the full source
+RUN mkdir -p /tmp/build/res/build
+COPY ./package*.json /tmp/build/
+WORKDIR /tmp/build
 RUN set -x && \
-    cd /tmp/build && \
     export PATH=$PWD/node_modules/.bin:$PATH && \
     npm install --loglevel http && \
-    npm pack && \
-    tar xzf devicefarmer-stf-*.tgz --strip-components 1 -C /app && \
-    bower cache clean && \
-    npm prune --production && \
-    mv node_modules /app && \
-    rm -rf ~/.node-gyp && \
-    mkdir /app/bundletool && \
-    mv /tmp/bundletool/* /app/bundletool && \
-    cd /app && \
-    find /tmp -mindepth 1 ! -regex '^/tmp/hsperfdata_root\(/.*\)?' -delete
-
-# Switch to the app user.
-USER stf
+    curl -sf https://gobinaries.com/tj/node-prune | sh
+
+wget --progress=dot:mega \
```

Comment:
The `wget `command has to be prefixed by the `RUN `command, otherwise the build is failing at the beginning